client/src/app/pages/applications/components/application-detail-drawer/components/download-button.tsx

```diff
@@ -1,56 +1,57 @@
-import React, { useState } from "react";
-import { Alert, Button } from "@patternfly/react-core";
+import React from "react";
+import { Button } from "@patternfly/react-core";
 import spacing from "@patternfly/react-styles/css/utilities/Spacing/spacing";
-import { Application } from "@app/api/models";
-import { Spinner } from "@patternfly/react-core";
-import { useDownloadStaticReport } from "@app/queries/download";
+import { saveAs } from "file-saver";
 
-export enum MimeType {
-  TAR = "tar",
-  YAML = "yaml",
+import { MimeType } from "@app/api/models";
+import { useFetchStaticReport } from "@app/queries/applications";
+import { NotificationsContext } from "@app/components/NotificationsContext";
+import { AxiosError } from "axios";
+import { getAxiosErrorMessage } from "@app/utils/utils";
+
+interface IDownloadButtonProps {
+  id: number;
+  mimeType: MimeType;
 }
-function DownloadButton({
-  application,
+
+export const DownloadButton: React.FC<IDownloadButtonProps> = ({
+  id,
   mimeType,
-}: {
-  application: Application;
-  mimeType: MimeType;
-}) {
-  const {
-    mutate: downloadFile,
-    isLoading,
-    isError,
-  } = useDownloadStaticReport();
+}) => {
+  const { pushNotification } = React.useContext(NotificationsContext);
+
+  const onFetchStaticReportError = (error: AxiosError) => {
+    pushNotification({
+      title: getAxiosErrorMessage(error),
+      variant: "danger",
+    });
+  };
+
+  const { data: report, refetch } = useFetchStaticReport(
+    id,
+    mimeType,
+    onFetchStaticReportError
+  );
 
   const handleDownload = () => {
-    downloadFile({
-      applicationId: application.id,
-      mimeType: mimeType,
+    refetch().then(() => {
+      if (report) {
+        saveAs(new Blob([report]), `analysis-report-app-${id}.${mimeType}`);
+      }
     });
   };
 
   return (
-    <>
-      {isLoading ? (
-        <Spinner size="sm" />
-      ) : isError ? (
-        <Alert variant="warning" isInline title={"Error downloading report"}>
-          <p>{"An error has occurred. Try to download again."}</p>
-        </Alert>
-      ) : (
-        <>
-          <Button
-            onClick={handleDownload}
-            id={`download-${mimeType}-button`}
-            variant="link"
-            className={spacing.pXs}
-          >
-            {mimeType === MimeType.YAML ? "YAML" : "Report"}
-          </Button>
-        </>
-      )}
-    </>
+    <Button
+      key={mimeType}
+      onClick={handleDownload}
+      id={`download-${mimeType}-button`}
+      variant="link"
+      className={spacing.pXs}
+    >
+      {mimeType === MimeType.YAML ? "YAML" : "Report"}
```

Comment:
Nit-pick: This text could just be passed in as a prop and set to whatever needed by the consuming component.

Reply:
@sjd78 , what do you mean ?
The text is determined from the mimeType prop already passed to the DownloadButton.
The text is then wrapped by the Button component corresponds to the Children prop.

Reply:
I mean you could put the regular text here:
https://github.com/konveyor/tackle2-ui/blob/c44218dd54457bd09de662c9580a56474f42f193/client/src/app/pages/applications/components/application-detail-drawer/application-detail-drawer-analysis.tsx#L137-L157

But then the button could look like:
```tsx
                      <DownloadButton
                        application={application}
                        mimeType={MimeType.TAR}
                      >
                        tar
                      </DownloadButton>
```

Reply:
I'm ok with the way it is now.  It's just a possible alternative.

Reply:
@sjd78 ,
Oh I see now, thanks for the example.
I don't see the gain, it would add extra lines of code for the same result.